tableauserverclient/models/tableau_auth.py

```diff
@@ -11,6 +12,8 @@ def __init__(self, username, password, site=None, site_id='', user_id_to_imperso
         self.password = password
         self.site_id = site_id
         self.username = username
+        self.token_name = token_name
```

Comment:
I believe there's logic where some of these can't be set at the same time, we should check in the __init__ and throw an exception for those cases (is it either username/password OR PAT?)

Reply:
The server will return an error if both are set with a clear message saying they are not allowed.
Do we want to replicate that logic in the client side? 
I haven't seen much validation going on in the models so I kept it out.

Reply:
The example I was thinking of was `connections` and `connection_credentials`, though I looked into it and that's validated in the endpoint, where it chooses which type of model to build, and not in the model itself.

If the error from Server is clear and caught correctly, that's sufficient thanks!

(The alternative I thought was to have 2 different auth objects, to keep this one from growing too large. Maybe that's not necessary now though)